### PR TITLE
fix: [PIE-2799]: Unable to create pipeline variable

### PR DESCRIPTION
**Pull request by yc147**
##### Jira Links:

https://harness.atlassian.net/browse/PIE-2799

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineVariablesContext/PipelineVariablesContext.tsx [hunk 1]
```diff
@@ -139,7 +139,9 @@ export function PipelineVariablesContextProvider(
   })
 
   React.useEffect(() => {
-    setResolvedPipeline(parse(defaultTo(resolvedPipelineResponse?.data?.mergedPipelineYaml, '')))
+    if (resolvedPipelineResponse?.data?.mergedPipelineYaml) {
+      setResolvedPipeline(parse(resolvedPipelineResponse.data.mergedPipelineYaml))
```
Comment (ishantmehtaharness) [inline src/modules/70-pipeline/components/PipelineVariablesContext/PipelineVariablesContext.tsx:143]: What if this is empty, we dont want to set empty?

Comment (ishantmehtaharness) [reply src/modules/70-pipeline/components/PipelineVariablesContext/PipelineVariablesContext.tsx:143]: Or we want to check, whether its defined or undefined?

Comment (yc147) [reply src/modules/70-pipeline/components/PipelineVariablesContext/PipelineVariablesContext.tsx:143]: The check will fail when it is empty

**Review by ishantmehtaharness — APPROVED** [review]
Looks good in this case
https://github.com/harness/harness-core-ui/pull/7456/files#r811875090

**Review by devinder-harness — APPROVED** [review]

**Review by vkbansal — APPROVED** [review]

**Pull request merged**